core/src/main/java/com/linecorp/armeria/server/Http2RequestDecoder.java

```diff
@@ -125,6 +126,11 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
                 return;
             }
 
+            final PathAndQuery pathAndQuery = PathAndQuery.parse(nettyHeaders.path().toString());
```

Comment:
Suggestion by @minwoox :

`PathAndQuery` can be passed to `ArmeriaHttpUtil.toArmeriaRequestHeaders` since it might be easier to test behavior since the logic is grouped in a separate method.